core/src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -122,11 +116,14 @@ object ConfigConvert extends LowPriorityConfigConvertImplicits {
     override def fromConfigObject(co: ConfigObject, default: Option[V] :: U): Try[FieldType[K, V] :: T] = {
       val keyStr = mapping(key.value.toString().tail)
       for {
-        v <- improveFailure(
-          vFieldConvert.value.from(co.get(keyStr)) match {
-            case Failure(CannotConvertNullException) =>
+        v <- improveFailure[V](
+          (co.get(keyStr), vFieldConvert.value) match {
+            case (null, converter: OptionConfigConvert[_]) =>
```

Comment:
It seems like the more open approach to this would be to introduce a trait, e.g. `AllowMissingConfigConvert` or `AllowNullConfigConvert`, which is used for the type check on `ConfigConvert.scala:121`.
Pureconfig would ship with `OptionConfigConvert` as the only child of that trait.